cmd/kubeadm/app/util/staticpod/utils_test.go

```diff
@@ -37,43 +40,114 @@ func TestComponentResources(t *testing.T) {
 
 func TestComponentProbe(t *testing.T) {
 	var tests = []struct {
-		port   int
-		path   string
-		scheme v1.URIScheme
+		name      string
+		cfg       *kubeadmapi.MasterConfiguration
+		component string
+		port      int
+		path      string
+		scheme    v1.URIScheme
+		expected  string
 	}{
 		{
-			port:   1,
-			path:   "foo",
-			scheme: v1.URISchemeHTTP,
+			name: "default apiserver advertise address with http",
+			cfg: &kubeadmapi.MasterConfiguration{
+				API: kubeadmapi.API{
+					AdvertiseAddress: "",
+				},
+			},
+			component: kubeadmconstants.KubeAPIServer,
+			port:      1,
+			path:      "foo",
+			scheme:    v1.URISchemeHTTP,
+			expected:  "127.0.0.1",
+		},
+		{
+			name: "default apiserver advertise address with https",
+			cfg: &kubeadmapi.MasterConfiguration{
+				API: kubeadmapi.API{
+					AdvertiseAddress: "",
+				},
+			},
+			component: kubeadmconstants.KubeAPIServer,
+			port:      2,
+			path:      "bar",
+			scheme:    v1.URISchemeHTTPS,
+			expected:  "127.0.0.1",
+		},
+		{
+			name: "valid ipv4 apiserver advertise address with http",
+			cfg: &kubeadmapi.MasterConfiguration{
+				API: kubeadmapi.API{
+					AdvertiseAddress: "1.2.3.4",
+				},
+			},
+			component: kubeadmconstants.KubeAPIServer,
+			port:      1,
+			path:      "foo",
+			scheme:    v1.URISchemeHTTP,
+			expected:  "1.2.3.4",
 		},
 		{
-			port:   2,
-			path:   "bar",
-			scheme: v1.URISchemeHTTPS,
+			name: "valid IPv4 scheduler probe",
+			cfg: &kubeadmapi.MasterConfiguration{
+				SchedulerExtraArgs: map[string]string{"address": "1.2.3.4"},
+			},
+			component: kubeadmconstants.KubeScheduler,
+			port:      1,
+			path:      "foo",
+			scheme:    v1.URISchemeHTTP,
+			expected:  "1.2.3.4",
+		},
+		{
+			name: "valid etcd probe using listen-client-urls IPv4 addresses",
```

Comment:
We should have here at least one test that would be able to handle IPv6 urls. E.g. for `::1` or something like `2001:db8::....`

Reply:
@kad thanks for the review. I'll add one or more IPv6 test cases and resubmit today.